Tools/CISupport/build-webkit-org/steps.py

```diff
@@ -459,8 +462,35 @@ def _addToLog(self, logName, message):
     def evaluateCommand(self, cmd):
         rc = super().evaluateCommand(cmd)
         steps_to_add = self.follow_up_steps()
-        if rc in (SUCCESS, WARNINGS) and self.getProperty('user_provided_git_hash'):
-            steps_to_add += [ArchiveMinifiedBuiltProduct(), UploadMinifiedBuiltProduct(), TransferToS3(terminate_build=True)]
+
+        triggers = self.getProperty('triggers', None)
+        if triggers:
+            # Build archive
+            steps_to_add += [ArchiveBuiltProduct()]
+            if CURRENT_HOSTNAME in BUILD_WEBKIT_HOSTNAMES + TESTING_ENVIRONMENT_HOSTNAMES:
+                steps_to_add.extend([
+                    GenerateS3URL(f"{self.getProperty('fullPlatform')}-{self.getProperty('architecture')}-{self.getProperty('configuration')}"),
+                    UploadFileToS3(f"WebKitBuild/{self.getProperty('configuration')}.zip", links={self.name: 'Archive'}),
+                ])
+            else:
+                # S3 might not be configured on local instances, achieve similar functionality without S3.
+                steps_to_add.extend([UploadBuiltProduct(), TransferToS3(terminate_build=True)])
+
+            # Minified build archive
+            full_platform = self.getProperty('fullPlatform')
+            if (full_platform.startswith('mac') or full_platform.startswith('ios-simulator') or full_platform.startswith('tvos-simulator')
```

Comment:
Nit: we can make this startswith call compact, as startswith can take a tuple. See: https://docs.python.org/3/library/stdtypes.html#str.startswith "prefix can also be a tuple of prefixes to look for".

e.g.: if (full_platform.startswith(('mac', 'ios-simulator', 'tvos-simulator', 'watchos-simulator'))